src/main/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorSpecificConfig.java

```diff
@@ -29,11 +29,20 @@
   @Description("The root dataset to browse when a root path is given")
   protected String rootDataset;
 
+  @Nullable
+  @Description("Option to show hidden datasets")
+  protected Boolean showHiddenDatasets;
+
   public BigQueryConnectorSpecificConfig(@Nullable String project, @Nullable String datasetProject,
                                          @Nullable String serviceAccountType, @Nullable String serviceFilePath,
                                          @Nullable String serviceAccountJson,
                                          @Nullable String rootDataset) {
     super(project, datasetProject, serviceAccountType, serviceFilePath, serviceAccountJson);
     this.rootDataset = rootDataset;
+    this.showHiddenDatasets = false;
+  }
+
+  public boolean showHiddenDatasets() {
+    return showHiddenDatasets == null ? false : showHiddenDatasets;
```

Comment:
I think we actually want to default to true since that's the more user friendly behavior, and let the upgrade back set this to false for any pre-existing connections.